pandas/tests/io/parser/common/test_common_basic.py

```diff
@@ -127,6 +127,27 @@ def test_1000_sep(all_parsers):
     tm.assert_frame_equal(result, expected)
 
 
+@xfail_pyarrow
+def test__search_replace_num_columns(all_parsers):
+    parser = all_parsers
+    data = """A;B;C
+155.75;3.001;43.06.4
+16.2;0;23.0
+"""
+    expected = DataFrame(
+        {
+            "A": ["155.75", "16.2"],
+            "B": [3001, 0],
+            "C": [43064, 230],
+        }
+    )
+
+    result = parser.read_csv(
+        StringIO(data), sep=";", dtype={"A": str, "B": int, "C": int}, thousands="."
```

Comment:
You should specify these explicitly

int is different on 32bit/windows compared to linux

Reply:
Thank you for your advice. I did specify int and that worked for me.

Seems like ci failures this time are not related to my changes.